### ISSUE TITLE
.NET Framework 4.7.2 Developer Pack installer crashes when run over WinRM

### ISSUE THREAD

**Issue by yaakov-h** [Microsoft/dotnet#839]
When running the [.NET Framework 4.7.2 Developer Pack](https://www.microsoft.com/net/download/thank-you/net472-developer-pack) installer (`NDP472-DevPack-ENU.exe`) over WinRM, the installer crashes. It does not crash when run from an interactive desktop session.

This can be reproduced by using Packer from Hashicorp, or Powershell Remoting, on a clean installation of Windows Server 2012 R2.

The installer exits with an exit code corresponding to 0xC0000005, which represents an Access Violation.

The only clue is a single entry in the Application event log:

```
Faulting application name: NDP472.exe, version: 4.7.3081.0, time stamp: 0x59074750
Faulting module name: wixstdba.dll, version: 3.11.1701.0, time stamp: 0x59074755
Exception code: 0xc0000005
Fault offset: 0x0000132e
Faulting process id: 0x4a4
Faulting application start time: 0x01d42eb0851f726b
Faulting application path: C:\Users\ADMINI~1\AppData\Local\Temp\{3A52AFB4-9AC4-4B14-969F-83984D4F2CFC}\.cr\NDP472.exe
Faulting module path: C:\Users\ADMINI~1\AppData\Local\Temp\{46D19905-2499-4730-B543-FD59D5BE394C}\.ba\wixstdba.dll
Report Id: 009dd28b-9aa4-11e8-80b4-00155dc51fc5
Faulting package full name: 
Faulting package-relative application ID: 
```

System event logs: [NDP472-diag-evtlogs.zip](https://github.com/Microsoft/dotnet/files/2309892/NDP472-diag-evtlogs.zip)

Contents of user's `%TEMP%` folder: [NDP472-diag-usertemp.zip](https://github.com/Microsoft/dotnet/files/2309886/NDP472-diag-usertemp.zip)

X-ref Microsoft Partner Support, case [REG:118080918754010].

See also: https://twitter.com/yaakov_h/status/1027050943441858560

**Comment by vivmishra** [comment]
@rakeshsinghranchi

**Comment by rakeshsinghranchi** [comment]
.NET 4.7.2 Developer pack installer is Burn based (WixToolSet) . Faulting module wixstdba.dll is embedded into wixbalextensions.dll of wixtoolset 3.11 . 

@barnson is this a known issue ?

**Comment by barnson** [comment]
@rakeshsinghranchi No, I haven't seen this before. Years ago, we had a bug when running via PSExec on certain OSes, but that's ancient history.

**Comment by jeannich** [comment]
I am facing this same issue on Windows 2008 R2

**Comment by rakeshsinghranchi** [comment]
Looking at the setup log shared by @yaakov-h  , .NET 4.7.2 runtime package chained by Burn bundle, could install successfully with pending reboot. 
```
Final Result: Installation completed successfully with success code: (0x80070BC2), "The requested operation is successful. Changes will not be effective until the system is rebooted. " (Elapsed time: 0 00:01:06).
```
Burn engine is throwing access violation exception and scenario is related to powershell remoting .
 @barnson , looks like this is the same issue as you mentioned about PSExec.

**Comment by barnson** [comment]
@rakeshsinghranchi: The PSExec bug was a hang, not a crash. This is something different. Needs debugging to determine why it's crashing during shutdown.

**Comment by jberezanski** [comment]
I can consistently reproduce the crash when installing the three latest developer packs (4.7.1, 4.7.2 and 4.8 preview) via PowerShell DSC on Windows Server 2016, Windows 10 1803 or Windows Server 2012 R2.

The faulting module is always wixstdba.dll. I can provide Process Monitor log files if needed.

Here are my repro steps: https://github.com/jberezanski/ChocolateyPackages/blob/devpack-crash/netfx-4.8-devpack/dscrepro.ps1

**Comment by rakeshsinghranchi** [comment]
Thanks @jberezanski . @barnson ,can you please help debug the issue? Also, any workaround will be helpful.

**Comment by barnson** [comment]
@rakeshsinghranchi, sorry, no, I'm not set up to repro. Suggest starting with a minidump. Symbols for 
v3.11.0.1701 are available in wix311-debug.zip at https://github.com/wixtoolset/wix3/releases/tag/wix311rtm.

**Comment by AlexWang-16** [comment]
I can confirm I am facing the same problem @jberezanski is having. I'm also on Windows Server 2016 using Ansible and win_chocolatey for installation. Everything works fine for v4.7

**Comment by vivmishra** [comment]
CC @NikolaMilosavljevic

**Comment by jeremygaither** [comment]
I can also confirm this is an issue installing 4.7.1 and 4.7.2 in a Packer build script over WinRM using Chocolatey package from @jberezanski. The installer exits with code `-1073741819` instead of `0` as logged in 
https://github.com/jberezanski/ChocolateyPackages/issues/22 as well. This happens regardless of rebooting before and after installation on a clean Windows Server 2016 image.

**Comment by lmunro** [comment]
@rakeshsinghranchi can we please have an update on this issue, it's been over a year and this is blocking automated builds of build agents that need this dev pack.

**Comment by solomonleaf** [comment]
Hi, is there a fix?  I am including NDP472-DevPack-ENU.exe as part of the WebPI feed we install and hitting the same issue.  My biggest concern is the installation is completed successfully when it returns C0000005.

**Comment by NikolaMilosavljevic** [comment]
@solomonleaf this was fixed in Wix toolset last year, see the linked issue and the PR: https://github.com/wixtoolset/wix3/pull/483

The fix was included in .NET 4.8 installer, which does not have this issue.

Yes, you can ignore this error - I'd suggest another way to verify that installation was successful, i.e. check for .NET Release registry key.

**Comment by yaakov-h** [comment]
FWIW, if you install the .NET 4.7.2 developer pack through the Visual Studio Installer, it does not have this issue. I don't know why.

**Comment by jeremygaither** [comment]
But there is no backported fix for the 4.7.2 dev pack installed over winrm? Does the 4.8 dev pack include targeting packs for 4.7.2?

**Comment by jeremygaither** [comment]
> FWIW, if you install the .NET 4.7.2 developer pack through the Visual Studio Installer, it does not have this issue. I don't know why.

@yaakov-h are you doing the install over winrm?

**Comment by yaakov-h** [comment]
@jeremygaither yes.

**Comment by jberezanski** [comment]
The [Chocolatey package](https://chocolatey.org/packages/netfx-4.7.2-devpack#releasenotes) works around this problem by recognizing that specific exit code and checking whether the SDK got actually installed. Perhaps the Visual Studio installer does something similar.

**Comment by txgsun** [comment]
Hey @NikolaMilosavljevic, right now I am using NDP471-DevPack-ENU.exe as part of WebPI feed, I am also receiving the exception 0xc0000005 but the installation can be completed. So for .NDP471-DevPack-ENU, does it like NDP472-DevPack-ENU we have mentioned before we wouldn't have a fix, since the fix was included in 4.8 installer. Besides, for this issue, can we just simply ignore it like NDP472? thx

**Comment by NikolaMilosavljevic** [comment]
> Hey @NikolaMilosavljevic, right now I am using NDP471-DevPack-ENU.exe as part of WebPI feed, I am also receiving the exception 0xc0000005 but the installation can be completed. So for .NDP471-DevPack-ENU, does it like NDP472-DevPack-ENU we have mentioned before we wouldn't have a fix, since the fix was included in 4.8 installer. Besides, for this issue, can we just simply ignore it like NDP472? thx

Yes, you can ignore this error for .NET 4.7.1 installation. I still suggest checking .NET Release key in registry to ensure the installation was indeed successful.
